pontoon/batch/actions.py

```diff
@@ -0,0 +1,200 @@
+
+    For documentation, refer to the `batch_action` function.
+
+    """
+    translations = translations.filter(approved=False)
+    changed_translation_pks = list(translations.values_list('pk', flat=True))
+
+    latest_translation_pk = None
+    if changed_translation_pks:
+        latest_translation_pk = translations.last().pk
+
+    count, translated_resources, changed_entities = get_translations_info(
+        translations,
+        locale,
+    )
+    translations.update(
+        approved=True,
+        approved_user=user,
+        approved_date=timezone.now(),
+        rejected=False,
+        rejected_user=None,
+        rejected_date=None,
+    )
+
+    return {
+        'count': count,
+        'translated_resources': translated_resources,
+        'changed_entities': changed_entities,
+        'latest_translation_pk': latest_translation_pk,
+        'changed_translation_pks': changed_translation_pks,
+    }
+
+
+def batch_reject_translations(form, user, translations, locale):
+    """Reject a series of translations.
+
+    Note that this function doesn't use the `translations` parameter, as it
+    needs to impact non-active translations. Hence it will generate its own
+    list of suggested translations to work on.
+
+    For documentation, refer to the `batch_action` function.
+
+    """
+    suggestions = Translation.objects.filter(
+        locale=locale,
+        entity__pk__in=form.cleaned_data['entities'],
+        approved=False,
+        rejected=False
+    )
+    count, translated_resources, changed_entities = get_translations_info(
+        suggestions,
+        locale,
+    )
+    TranslationMemoryEntry.objects.filter(translation__in=suggestions).delete()
+    suggestions.update(
+        rejected=True,
+        rejected_user=user,
+        rejected_date=timezone.now(),
+        approved=False,
+        approved_user=None,
+        approved_date=None,
+    )
+
+    return {
+        'count': count,
+        'translated_resources': translated_resources,
+        'changed_entities': changed_entities,
+        'latest_translation_pk': None,
+        'changed_translation_pks': [],
+    }
+
+
+def batch_replace_translations(form, user, translations, locale):
+    """Replace characters in a series of translations.
+
+    Replaces all occurences of the content of the `find` parameter with the
+    content of the `replace` parameter.
+
+    For documentation, refer to the `batch_action` function.
+
+    """
+    find = form.cleaned_data['find']
+    replace = form.cleaned_data['replace']
+
+    try:
+        translations, changed_translations = translations.find_and_replace(
+            find,
+            replace,
+            user
+        )
+        translations.update(
+            approved=False,
+            approved_user=None,
+            approved_date=None,
+            rejected=True,
+            rejected_user=user,
+            rejected_date=timezone.now(),
+            fuzzy=False,
+        )
```

Comment:
Note that `translations` only contains "active" translations; I've added another comment where `translations` are collected. That means non-active suggestions stay unreviewed after search & replace.

I suggest we use this refactor to get more specific in variable names.

Reply:
As we figured out, this is actually a feature - we don't want people to reject translations they haven't reviewed yet.